### PR TITLE
Feature/hasndling user exceptions epmdj 10676

### PR DESCRIPTION
**Pull request by iryabov**

### CODE REVIEW

File: admin-auth/src/test/kotlin/com/epam/drill/admin/auth/UserAuthenticationTests.kt [hunk 1]
```diff
@@ -134,7 +133,98 @@ class UserAuthenticationTest {
...
+                val form = RegistrationPayload(username = "guest", password = "secret")
+                setBody(Json.encodeToString(RegistrationPayload.serializer(), form))
+            }) {
+                assertEquals(HttpStatusCode.BadRequest, response.status())
+            }
+        }
+    }
+
+    @Test
+    fun `given without authentication 'POST update-password' must fail with 401 status`() {
```
Comment (RomanDavlyatshin) [inline admin-auth/src/test/kotlin/com/epam/drill/admin/auth/UserAuthenticationTests.kt]: `given without authentication...` is grammatically incorrect. Remove word `given`

File: admin-auth/src/test/kotlin/com/epam/drill/admin/auth/TestApplication.kt [hunk 2]
```diff
@@ -18,13 +18,33 @@ package com.epam.drill.admin.auth
...
 import org.kodein.di.ktor.di
 
-fun testApp(
+
+fun TestApplicationEngine.withStatusPages() {
+    callInterceptor = {
+        try {
+            call.application.execute(call)
+        } catch (_: Throwable) {
+            //allow to handle status pages
```
Comment (RomanDavlyatshin) [inline admin-auth/src/test/kotlin/com/epam/drill/admin/auth/TestApplication.kt:37]: _tbd_

File: admin-auth/src/main/kotlin/com/epam/drill/admin/auth/route/UserAuthenticationRoutes.kt [hunk 3]
```diff
@@ -42,6 +46,20 @@ object UpdatePassword
...
+    }
+    exception<IncorrectCredentialsException> { cause ->
+        logger.trace(cause) { "401 Username or password is incorrect" }
+        call.respond(HttpStatusCode.Unauthorized, "Username or password is incorrect")
+    }
+    exception<UserValidationException> { cause ->
+        logger.trace(cause) { "400 User data is invalid" }
+        call.respond(HttpStatusCode.BadRequest, "User data is invalid")
+    }
+}
```
Comment (RomanDavlyatshin) [inline admin-auth/src/main/kotlin/com/epam/drill/admin/auth/route/UserAuthenticationRoutes.kt:62]: can we add exception<Throwable> as a last clause that will handle or other exception types and'll set HTTP 500 code?

**Review by RomanDavlyatshin — CHANGES_REQUESTED** [review]

File: admin-auth/src/main/kotlin/com/epam/drill/admin/auth/route/UserAuthenticationRoutes.kt (hunk 3 above)
Comment (RomanDavlyatshin) [reply admin-auth/src/main/kotlin/com/epam/drill/admin/auth/route/UserAuthenticationRoutes.kt:62]: add message + data + errors into Exception classes and pass to response in exception<MyException> handling

Comment (iryabov) [reply admin-auth/src/main/kotlin/com/epam/drill/admin/auth/route/UserAuthenticationRoutes.kt:62]: will be fixed in the next branch

**Pull request merged**